src/encode/xz.rs

```diff
@@ -28,25 +24,30 @@ where
     let index_size = write_index(output, unpadded_size, unpacked_size)?;
 
     // Footer
-    write_footer(output, flags, index_size)
+    write_footer(output, check_type, index_size)
 }
 
-fn write_header<W>(output: &mut W, flags: u16) -> io::Result<()>
+fn write_header<W>(output: &mut W, check_type: header::CheckMethod) -> io::Result<()>
 where
     W: io::Write,
 {
-    output.write_all(XZ_MAGIC)?;
+    output.write_all(header::XZ_MAGIC)?;
     let mut digest = crc32::Digest::new(crc32::IEEE);
     {
         let mut digested = util::HasherWrite::new(output, &mut digest);
-        digested.write_u16::<BigEndian>(flags)?;
+        digested.write_u8(0)?;
```

Comment:
This zero seems weird here. I think we should keep some `Flags` abstraction, which can be created from a `CheckMethod` (for now we don't use other flags), and converted into a `u16` (we may want to add other flags in the future).